bin/download-prebuilt-firmware.py

```diff
@@ -3,24 +3,59 @@
 # FIXME: Make this work under Python 2
 
 import argparse
-from collections import namedtuple
 import csv
 import doctest
 import json
 import os
+import pickle
 import sys
 import time
 import urllib.request
 
+from collections import namedtuple
+from datetime import datetime
 
-def ls_github(url):
-    while True:
-        data = json.loads(urllib.request.urlopen(url).read().decode())
-        if "message" in data:
-            print("Warning: {}".format(data["message"]))
-            time.sleep(1)
-            continue
-        return data
+
+class TargetNotFound(Exception):
+    pass
+
+
+def ls_github(url, cache_ttl=None):
+
+    # FIXME: Move cache to a class or find a package.
+    cache_name = "github.pickle"
+
+    def load_cache():
+        try:
+            cache = pickle.load(open(cache_name, 'rb'))
+        except IOError:
+            cache = {}
+        return cache
+
+    def save_cache(cache):
+        pickle.dump(cache, open(cache_name, 'wb'))
+
+    cache = load_cache()
+    if url in cache and (cache_ttl is None or (
+        (datetime.now()-cache[url]['timestamp']).total_seconds() <
+            cache_ttl)):
+        data = cache[url]['data']
+    else:
+        while True:
+            data = json.loads(urllib.request.urlopen(url).read().decode())
+            if "message" in data:
+                print("Warning: {}".format(data["message"]))
+                time.sleep(1)
+                continue
+            else:
+                break
```

Comment:
Please format the `cache[url]` dictionary more normally. I think this is pep8 compliant?
```python
       cache[url] = {
           'timestamp': datetime.now(),
           'data': data,
       }
```